### Exercício 01 – Manipulando dataframes com Pandas.
        a) Abra o arquivo world_happiness_report_2015.csv e o aloque em um dataframe.
        b) Verifique o cabeçalho e o final do dataframe.
        c) Quais as colunas desse dataframe?
        d) Quais os tipos de dados temos no dataframe?
        e) Há valores faltantes ou nulos? Em quais colunas?
        f) Renomeie as variáveis como segue:

In [20]:
import pandas as pd 

caminho = ("C:\\Users\\FIC\\Documents\\gustavo-muniz-python\\Arquivos\\world_happiness_report_2015.csv")

df = pd.read_csv (caminho)

df.head(2)

country          region  happiness rank  happiness score  \
0  Switzerland  Western Europe               1            7.587   
1      Iceland  Western Europe               2            7.561   

   standard error  economy (GDP per Capita)   family  \
0         0.03411                   1.39651  1.34951   
1         0.04884                   1.30232  1.40223   

   health (Life Expectancy)  freedom  trust (Government Corruption)  \
0                   0.94143  0.66557                        0.41978   
1                   0.94784  0.62877                        0.14145   

   generosity  dystopia residual  
0     0.29678            2.51738  
1     0.43630            2.70201

In [21]:
df.tail(1)

country              region  happiness rank  happiness score  \
163    Togo  Sub-Saharan Africa             158            2.839   

     standard error  economy (GDP per Capita)   family  \
163         0.06727                   0.20868  0.13995   

     health (Life Expectancy)  freedom  trust (Government Corruption)  \
163                   0.28443  0.36453                        0.10731   

     generosity  dystopia residual  
163     0.16681                NaN

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 164 entries, 0 to 163
Data columns (total 12 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   country                        164 non-null    object 
 1   region                         164 non-null    object 
 2   happiness rank                 164 non-null    int64  
 3   happiness score                156 non-null    float64
 4   standard error                 163 non-null    float64
 5   economy (GDP per Capita)       164 non-null    float64
 6   family                         164 non-null    float64
 7   health (Life Expectancy)       164 non-null    float64
 8   freedom                        164 non-null    float64
 9   trust (Government Corruption)  164 non-null    float64
 10  generosity                     164 non-null    float64
 11  dystopia residual              150 non-null    float64
dtypes: float64(9), int64(1), object(2)
memory usage: 1

In [18]:
df = df.rename(columns={
    'happiness rank':'rank_felicidade',
    'hapiness score': 'score_felicidade',
    'standart error':'stand_error',
    'economy (GDP per Capita)': 'PIB',
    'health (Life Expectancy)':'expect_vida',
    'trust (Government Corruption)':'corrupcao'
})

df.head()

country          region  rank_felicidade  happiness score  \
0  Switzerland  Western Europe                1            7.587   
1      Iceland  Western Europe                2            7.561   
2      Denmark  Western Europe                3            7.527   
3       Norway  Western Europe                4            7.522   
4       Canada   North America                5            7.427   

   standard error      PIB   family  expect_vida  freedom  corrupcao  \
0         0.03411  1.39651  1.34951      0.94143  0.66557    0.41978   
1         0.04884  1.30232  1.40223      0.94784  0.62877    0.14145   
2         0.03328  1.32548  1.36058      0.87464  0.64938    0.48357   
3         0.03880  1.45900  1.33095      0.88521  0.66973    0.36503   
4         0.03553  1.32629  1.32261      0.90563  0.63297    0.32957   

   generosity  dystopia residual  
0     0.29678            2.51738  
1     0.43630            2.70201  
2     0.34139            2.49204  
3     0.34699            2.46531  
4     0.45811            2.45176

### Respostas 

D - Temos : STR, Inteiro e Float

E - Sim, temos valores faltntes. Na coluna "happiness score" temos 156, "standard error" temos 163 e "dystopia residual" temos 150. Não temos valores nulos.

In [19]:
df.describe(include='all')

country              region  rank_felicidade  happiness score  \
count         164                 164       164.000000       156.000000   
unique        158                  10              NaN              NaN   
top     Venezuela  Sub-Saharan Africa              NaN              NaN   
freq            2                  40              NaN              NaN   
mean          NaN                 NaN        77.018293         5.424487   
std           NaN                 NaN        46.699860         1.212568   
min           NaN                 NaN         1.000000         2.000000   
25%           NaN                 NaN        35.750000         4.561250   
50%           NaN                 NaN        76.500000         5.260500   
75%           NaN                 NaN       117.250000         6.422000   
max           NaN                 NaN       158.000000         7.587000   

        standard error         PIB      family  expect_vida     freedom  \
count       163.000000  164.000000  164.000000   164.000000  164.000000   
unique             NaN         NaN         NaN          NaN         NaN   
top                NaN         NaN         NaN          NaN         NaN   
freq               NaN         NaN         NaN          NaN         NaN   
mean          0.047692    0.862023    1.000812     0.638187    0.434959   
std           0.017110    0.404519    0.272449     0.246327    0.152201   
min           0.018480    0.000000    0.000000     0.000000    0.000000   
25%           0.036780    0.579618    0.863468     0.460545    0.331247   
50%           0.043940    0.920510    1.038145     0.705930    0.440745   
75%           0.052335    1.190585    1.226987     0.814975    0.557190   
max           0.136930    1.690420    1.402230     1.025250    0.669730   

         corrupcao  generosity  dystopia residual  
count   164.000000  164.000000         150.000000  
unique         NaN         NaN                NaN  
top            NaN         NaN                NaN  
freq           NaN         NaN                NaN  
mean      0.149978    0.239615           2.128909  
std       0.124201    0.128053           0.563160  
min       0.000000    0.000000           0.328580  
25%       0.063010    0.152017           1.811778  
50%       0.110690    0.217155           2.120700  
75%       0.189380    0.319570           2.474890  
max       0.551910    0.795880           3.602140

### G -     
        O valor médio da expectativa de vida é 0.638187	
        O valor mediano da expectativa de vida é 0.705930	
        O TOP da variavel PIB é 1.690420